js/assessments/paragraphTooLongAssessment.js

```diff
@@ -0,0 +1,86 @@
+var AssessmentResult = require( "../values/AssessmentResult.js" );
+var fixFloatingPoint = require( "../helpers/fixFloatingPoint.js" );
+var filter = require( "lodash/filter" );
+
+/**
+ * The function to use as a filter for too long paragraphs.
+ * @param {number} recommendedValue The recommended maximum length of a paragraph.
+ * @param {number} paragraphLength The number of words within a paragraph.
+ * @returns {boolean} Returns true if paragraphLength exceeds recommendedValue.
+ */
+var isParagraphTooLong = function( recommendedValue, paragraphLength ) {
+	return paragraphLength > recommendedValue;
+};
+
+/**
+ * Returns an array containing only the paragraphs longer than the recommended length.
+ * @param {array} paragraphsLength The array containing the lengths of individual paragraphs.
+ * @param {number} recommendedValue The recommended maximum length of a paragraph.
+ * @returns {number} The number of too long paragraphs.
+ */
+var getTooLongParagraphs = function( paragraphsLength, recommendedValue ) {
+	var tooLongParagraphs = filter( paragraphsLength, isParagraphTooLong.bind( null, recommendedValue ) );
+	return tooLongParagraphs.length;
+};
+
+/**
+ * Returns the scores and text for the ParagraphTooLongAssessment
+ * @param {array} paragraphsLength The array containing the lengths of individual paragraphs.
+ * @param {number} tooLongParagraphs The number of too long paragraphs.
+ * @param {number} recommendedValue The recommended maximum length of a paragraph.
+ * @param {object} i18n The i18n object used for translations.
+ * @returns {{score: number, text: string }} the assessmentResult.
+ */
+var calculateParagraphLengthResult = function( paragraphsLength, tooLongParagraphs, recommendedValue, i18n ) {
+	// 6 is the number of scorepoints between 3, minscore and 9, maxscore. For scoring we use 100 steps, each step is 0.06.
+	// Up to 117 is for scoring a 9, higher numbers give lower scores.
+	// floatingPointFix because of js rounding errors
+	var score = 9 - Math.max( Math.min( ( 0.06 ) * ( paragraphsLength[ 0 ] - 117 ), 6 ), 0 );
+	score = fixFloatingPoint( score );
+	if ( score >= 7 ) {
+		return {
+			score: score,
+			text: i18n.dgettext( "js-text-analysis", "None of your paragraphs is too long, which is great." )
```

Comment:
I'd vouch for replacing `is` with `are`. Both are grammatically correct, but seeing as we're talking about multiple paragraphs, `are` feels more natural.